Civi/Funding/FundingCase/StatusDeterminer/DefaultFundingCaseStatusDeterminer.php

```diff
@@ -47,7 +49,16 @@ public function __construct(
   }
 
   public function getStatus(string $currentStatus, string $action): string {
-    return 'approve' === $action ? 'ongoing' : $currentStatus;
+    switch ($action) {
+      case FundingCaseActions::APPROVE:
+        return FundingCaseStatus::ONGOING;
+
+      case FundingCaseActions::FINISH_CLEARING:
+        return FundingCaseStatus::CLEARED;
+
```

Comment:
why is APPROVE and FINISH_CLEARING being handled differently? is this a state transititon?

Reply:
If the action applied to the funding case is `FundingCaseActions::FINISH_CLEARING`, the new funding case status will be `FundingCaseStatus::CLEARED`. If the applied action is `FundingCaseActions::APPROVE` the new status will be `FundingCaseStatus::ONGOING`.